test/Grpc.Net.Client.Tests/Balancer/ConnectionManagerTests.cs

```diff
@@ -327,18 +327,139 @@ public async Task UpdateAddresses_ConnectIsInProgress_InProgressConnectIsCancele
             Assert.AreEqual(81, connectAddress2.Port);
         }
 
+        [Test]
+        public async Task PickAsync_DoesNotDeadlockAfterReconnect_WithResolverError()
+        {
+            // Arrange
+            var services = new ServiceCollection();
+            services.AddNUnitLogger();
+            await using var serviceProvider = services.BuildServiceProvider();
+            var loggerFactory = serviceProvider.GetRequiredService<ILoggerFactory>();
+
+            var resolver = new TestResolver(loggerFactory);
+
+            GrpcChannelOptions channelOptions = new GrpcChannelOptions();
+            channelOptions.ServiceConfig = new ServiceConfig()
+            {
+                LoadBalancingConfigs = { new RoundRobinConfig() }
+            };
+
+            var transportFactory = new TestSubchannelTransportFactory(async (_, ct) =>
+            {
+                await Task.Delay(1, ct);
```

Comment:
Does this need to be a delay? Can you just `Task.FromResult(...)` or even `Task.Yield()` if you really need to go async.